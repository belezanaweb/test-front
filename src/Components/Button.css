.btn {
  font-family: "Helvetica Neue", "Helvetica", "Arial", "sans-serif";
  font-weight: 600;
  font-size: calc(16px + 1vmin);

  text-transform: uppercase;

  background: #ff6a00;
  color: #ffffff;

  width: 100%;
  height: 60px;

  border: 0;
  border-radius: 4px;

  box-shadow: #d45a00 0 3px 0, 1px 5px 3px rgba(35, 26, 26, 0.25);

  margin-top: 10px;
  margin-bottom: 10px;
}

.btn:hover {
  background: rgb(212, 90, 0);
  box-shadow: unset;
  border: unset;
}
