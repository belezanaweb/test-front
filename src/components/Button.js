import React from 'react'; 
import styled from 'styled-components'; 

const Buttons = styled.button`
    height: 60px;
    width: 341px;
    border-radius: 3px;
    background-color: #FF6C00;
    box-shadow: inset 0 -3px 0 0 #D45A00, 0 2px 4px 0 rgba(0,0,0,0.25);
    margin-top: 3.5vh;
`
const Name = styled.div`
    height: 24px;
    /* width: 339px; */
    color: #FFF;
    font-family: "Helvetica Neue";
    font-size: 20px;
    font-weight: 700;
    letter-spacing: 0.5px;
    line-height: 24px;
    text-align: center;
`

const Button = (props) => {
    return (
        <Buttons>
            <Name>
                {props.children}
            </Name>
        </Buttons>
    )
}

export default Button;