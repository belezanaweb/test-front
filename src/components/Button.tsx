import React from 'react'
import { useHistory } from 'react-router-dom'

import { styled, keyframes } from '@/style/stitches.config'

const onHover = keyframes({
  '0%': { backgroundColor: '$orange100' },
  '100%': { backgroundColor: '$orange300' }
})

const ButtonStyled = styled('button', {
  backgroundColor: '$orange100',
  width: '-webkit-fill-available',
  color: 'white',
  fontSize: 20,
  fontWeight: 'bold',
  textTransform: 'uppercase',
  marginX: '10px',
  marginY: '5px',
  paddingY: '15px',
  paddingX: '15px',
  border: 'none',
  borderBottom: '4px solid $orange300',
  borderRadius: '3px',
  '&:hover': {
    animation: `${onHover} 300ms`,
    backgroundColor: '$orange300'
  }
})

interface OwnProps {
  text: string
  href?: string
  isSubmit?: boolean
}

const Box = (props: OwnProps): React.ReactElement => {
  const { text, href, isSubmit = false } = props

  const history = useHistory()

  const onClick = (): void => {
    if (href) {
      history.push(href)
    }
  }

  if (isSubmit) {
    return (
      <div>
        <ButtonStyled type="submit" data-testid="button">
          {text}
        </ButtonStyled>
      </div>
    )
  }

  return (
    <ButtonStyled onClick={href ? onClick : undefined} data-testid="button">
      {text}
    </ButtonStyled>
  )
}

export default Box
