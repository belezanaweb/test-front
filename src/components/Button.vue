<template>
  <div>
    <div class="row">
      <div class="col-xs-12 col-md-6 offset-md-3">
        <button
          type="button"
          class="btn btn-lg btn-block btn-danger text-uppercase font-weight-bold mb-3 button"
          @click="$emit('click')"
        >
          {{ title }}
        </button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Button',
  props: {
    title: {
      type: String,
      default: '',
    },
  },
};
</script>

<style scoped lang="scss">
.button {
  padding: 1.071rem 0;
}
.btn-danger {
  background-color: #ff7800;
  border-color: #ff7800;
  &:hover {
    background-color: #d45a00;
    border-color: #d45a00;
  }
}
</style>
