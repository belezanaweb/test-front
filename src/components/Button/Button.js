import React from 'react'

const Button = () => {
  return (
    <div>
      <button></button>
    </div>
  )
}

export default Button
