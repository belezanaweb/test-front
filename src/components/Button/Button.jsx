import React from 'react'

import { StyledButton } from './styles'

function Button({ children, ...rest }) {
  return <StyledButton {...rest}>{children}</StyledButton>
}

export default Button
