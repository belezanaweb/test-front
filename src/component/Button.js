import React from 'react'
import * as S from './Button.styles'

const Button = ({ text, onClick, ...props }) => {
  return (
    <S.StyledButton onClick={onClick} {...props}>
      {text}
    </S.StyledButton>
  )
}

export default Button
