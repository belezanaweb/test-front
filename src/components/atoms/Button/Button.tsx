import * as Styles from './Button.styles'
interface ButtonProps {
  text: string;
  type: ('button' | 'submit');
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ text, disabled = false, type, onClick }) => {
  return (
    <Styles.Container>
      <Styles.Button type={type} disabled={disabled} onClick={onClick}>
        {text}
      </Styles.Button>
    </Styles.Container>

  )
}

export default Button
