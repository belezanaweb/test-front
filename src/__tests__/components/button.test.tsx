import React from 'react'
import { render, screen } from '../../util/test-utils'
import Button from '../../components/Button'

describe('Button Component', () => {
  it('should be able to render a button', () => {
    render(<Button>Buy now</Button>)

    expect(screen.getByRole('button', { name: /Buy now/i })).toBeInTheDocument()
  })

  it('should render a disabled button', () => {
    render(<Button disabled>Buy now</Button>)

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyle({
      cursor: 'not-allowed'
    })
  })

  it('should render button as a link', () => {
    render(
      <Button as="a" href="/link">
        Buy now
      </Button>
    )

    expect(screen.getByRole('link', { name: /buy now/i })).toHaveAttribute(
      'href',
      '/link'
    )
  })
})
