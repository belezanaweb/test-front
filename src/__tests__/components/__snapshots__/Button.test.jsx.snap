// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match with snapshot 1`] = `
<div>
  <button
    class="sc-bdnxRM grNhhY"
  >
    Button Child
  </button>
</div>
`;
